zeus/device/soc/apple.py

```diff
@@ -0,0 +1,154 @@
+"""Apple Silicon SoC's."""
+
+from __future__ import annotations
+
+import sys
+import platform
+from dataclasses import dataclass, asdict, fields
+
+from zeus.device.soc.common import SoC, SoCMeasurement, ZeusSoCInitError
+
+# The following are optional dependencies. If a host machine does not have them
+# installed, the Zeus code importing this module will gracefully handle the
+# import error.
+from zeus_apple_silicon import AppleEnergyMonitor, AppleEnergyMetrics  # type: ignore
+
+
+class ZeusAppleInitError(ZeusSoCInitError):
+    """Import error for Apple SoC initialization failures."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+
+@dataclass
+class AppleSiliconMeasurement(SoCMeasurement):
+    """Represents energy consumption of various subsystems on an Apple processor.
+
+    All measurements are in mJ.
+    """
+
+    # CPU related metrics
+    cpu_total_mj: int | None = None
+    efficiency_cores_mj: list[int] | None = None
+    performance_cores_mj: list[int] | None = None
+    efficiency_core_manager_mj: int | None = None
+    performance_core_manager_mj: int | None = None
+
+    # DRAM
+    dram_mj: int | None = None
+
+    # GPU related metrics
+    gpu_mj: int | None = None
+    gpu_sram_mj: int | None = None
+
+    # ANE (Apple Neural Engine)
+    ane_mj: int | None = None
+
+    def __sub__(self, other: "AppleSiliconMeasurement") -> "AppleSiliconMeasurement":
```

Comment:
```suggestion
    def __sub__(self, other: AppleSiliconMeasurement) -> AppleSiliconMeasurement:
```

You've already imported `__future__.annotations`, you don't need quotes. Applies to all other type annotations.